src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -662,6 +741,7 @@ internal CachedArrayField(string name, Cci.INamedTypeDefinition containingType,
         }
 
         public override ImmutableArray<byte> MappedData => default(ImmutableArray<byte>);
+        public override bool IsReadOnly => false;
```

Comment:
:memo: Note this affects the caching field for blob wrappers too (existing logic, see `tryEmitAsCachedArrayFromBlob` which involves an assignment to the caching field, so it cannot be read only).